test/crowdsale/PremintedCrowdsale.test.js

```diff
@@ -0,0 +1,67 @@
+import ether from '../helpers/ether';
```

Comment:
Not sure about the name "PremintedCrowdsale" here...
In a way, the base Crowdsale is also preminted, because it assumes that it has a positive balance for the associated token. In this case, we also assume that the token has been pre-minted, AND assume that the crowdsale has approval/allowance.

Reply:
We had disscussed this with @facuspagnuolo, since ApprovedCrowdsale wasn't too explanatory.. But I fully agree with your comment that the base is also preminted. Any other proposal?

Reply:
Some ideas:
DelegateCrowdsale, ProxyCrowdsale, ForwardingCrowdsale, AuthorizedCrowdsale?

Reply:
AllowanceCrowdsale?

Reply:
LGTM